src/lib/shadow-shim-helper-rs/src/scmutex.rs

```diff
@@ -31,67 +31,101 @@ mod sync {
     }
 
     #[cfg(not(loom))]
-    pub unsafe fn futex(
-        futex_word: &AtomicI32,
-        futex_op: i32,
-        val: i32,
-        timeout: *const libc::timespec,
-        uaddr2: *mut i32,
-        val3: i32,
+    pub(super) unsafe fn futex_wait(
+        futex_word: &super::AtomicFutexWord,
+        val: super::FutexWord,
     ) -> nix::Result<i64> {
+        // The kernel just checks for equality of the value at this pointer.
+        // Just validate that it's the right size and alignment.
+        assert_eq!(
+            std::mem::size_of::<super::AtomicFutexWord>(),
+            std::mem::size_of::<u32>()
+        );
+        assert_eq!(
+            std::mem::align_of::<super::AtomicFutexWord>(),
+            std::mem::align_of::<u32>()
+        );
+        let futex_word: *const u32 = futex_word as *const _ as *const u32;
```

Comment:
Maybe add this as a `as_u32` method to `AtomicFutexWord` that is only available if `#[cfg(not(loom))]`?